### ISSUE TITLE
problem in categorical_focal_loss

### ISSUE THREAD

**Issue by LeeJayHui** [qubvel/segmentation_models#309]
Hi, I think there is probably a mistake in here:
https://github.com/qubvel/segmentation_models/blob/a6fbf2d5ea21081929df634d095d389211e649ef/segmentation_models/base/functional.py#L278.
The values in gt array here are either one or zero. The above implementation would just simply ignore the zero ground truth when calculating the focal loss. I think maybe it should be written as follow (alpha do nothing here) :
- pr = backend.clip(pr, backend.epsilon(), 1.0 - backend.epsilon())

- pr = tf.where(tf.greater(gt, 0.), pr, 1-pr)

- loss = - backend.pow((1 - pr), gamma) * backend.log(pr)
